### DISCUSSION TITLE
How to pass arguments to a service's __construct() method when autowiring it to a controller?

### DISCUSSION THREAD

**Discussion by VitaliyMarkelov** [symfony/symfony#57448] [Q&A]
Say I have a service
```
namespace App\Service;
class MyService {
    public function __construct () {}
   // .....
    public function myServiceMethod() {...}
}
```

and a controller:
```
namespace App\Controller;
use App\Service\MyService
class MyController extends AbstractController {

    public function __construct(
        private MyService $myService,
    )
    {
       public function MyControllerMethod () {
       ...
       $this->myService->myServiceMethod();
       ...
       }
    }
}
```
and these work perfectly OK. In my case I would like to pass init parameters to my service, so I changed constructor:

```
namespace App\Service;
class MyService {
    public function __construct (string $init) {
        $this->init = $init;
    }
    ......
}

```
and get error. How can I pass an argument to MyService constructor? May be this just a wrong (non-symfonysh) approach and/or bad coding habit, then I would love to get an advice on a good approach. If there a doc that discusses this it would be just wonderful. I have watched and rewatched SymfonyCasts Creating Your Own Service from Cosmic Coding with Symfony 7 to no avail ).

**Comment by chapterjason** [comment]
Hey, please have a look at the documentation.

You can either use the service definition and pass it in the yaml/xml/php `arguments` as `%param_name%` (like here https://symfony.com/doc/current/service_container.html#manually-wiring-arguments) or you can use the `Autowire`-Attribute for that (https://symfony.com/doc/current/service_container/autowiring.html#fixing-non-autowireable-arguments)
